Updates for 3.15.2 branch

Eric-revollo opened the pull request:
Updates that go to the  3.15.2 branch, please check before we merge to 3.15.2 branch

niharsawant commented:
## FusionCharts
* FC3: 
    * In case of change log, it is applicable for more than 70+ charts not just Marimekko 
    * Use better dataset in changed behaviour. @sikrigagan please help with the sameple
* FC4: `getting-started/plain-javascript/css-transformations.md` should be under `Configuring Charts / Common Usecases`. Also, there should be some examples explaining the feature and API
* FC5: Sample not working when `useZoomLine` is set to `0`.
* FC7: Missing in change log

[Note: Yet to review rest of the items. Will be adding more details]

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+- Decimal and thousand separator attributes now work properly on Marimekko chart. 
+
+- For all 2D and 3D bar chart variants, the fontSize attribute now works properly when set with a numeric value. Earlier a JS error was thrown.
```
I think it should be `For all 2D and 3D column and bar chart variants, the `fontSize` and `labelFontSize` attributes now work properly when set to a string and numeric value. Earlier a JS error was thrown when numeric value was set.

Eric-revollo replied:
changed

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
 
+<h4>Fixes</h4>
+
+- In Marimekko chart, using `yAxisValueDecimal` now works properly when `setAdaptiveYMin` and `forceYAxisValueDecimals` are enabled.
```
This fix is valid for more than 70+ chart types with cartesian axis, not just the marimekko chart. Also the fix description can be better something link: `For cartesian axis charts, `yAxisValueDecimal` attribute now works properly when `setAdaptiveYMin` and `forceYAxisValueDecimals` attributes are enabled.`

Eric-revollo replied:
changed

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+- Using the `isRaw` boolean property `getJSONData()` and `getChartData()` now return property names in the same format as defined in the dataSource.
+
+- The space between legend items has been decreased in order to provide an improved visualization of data plots. 
```
```suggestion
```

Eric-revollo replied:
This one should be meant for FC 8: Spacing concerns in legend items

sikrigagan replied:
Actually for FC 8 the correct description would be: `Going by the principles of data visualisation we have improved the data ink ratio for legends in charts to avoid unexpected space between legend items.`

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+<h4>Improvements</h4>
+
+- The space between legend items has been decreased in order to provide an improved visualization of data plots. 
```
Data plot and legends are two different components on chart canvas and this issue was specifically for IE browser and we should mention the same.
```suggestion
- The spacing between legend items is improved now for IE 11. Previously, the spacing was little peculiar as white-space was being considered as a line break and `<br>` tag was considered twice.
```

The space between legend items has been decreased in order to provide an improved visualization of data plots.

Eric-revollo replied:
according to PRD suggested text looks like  for FC 7: Legend item alignment issue in IE 11 browser

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
 
 <p class="release-date">2020</p>
 
+<h4>Fixes</h4>
```
- FC 8 is missing

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+- The space between legend items has been decreased in order to provide an improved visualization of data plots. 
+
+- When exporting spark charts to a CSV file, the exported csv data displays now displays column headers. Previously, the header was not included in the CSV file.
```
```suggestion
- When exporting spark charts as a CSV file or CSV data using `getCSVData()` or `getChartData('csv')` methods, the exported CSV now displays column headers. Previously, the column headers were not included.
```

Eric-revollo replied:
changed

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
 
 <p class="release-date">2020</p>
 
```
New Feature section is missing - It should be present and FC 12 - Events support in chord diagram should be included there.

Eric-revollo replied:
Added

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+- The event handlers `chartRollOver` and `chartRollOut`are now working properly when multiple charts are present on a web page. Earlier, upon multiple instances of hover over each other one of the charts stopped invoking the hover events and threw a JS error. 
+
+- When using stacked charts, the macros `$sum` and `$unformattedSum` are now working properly for customized tooltips.Previously, the `$sum` and `$unformattedSum` macros were returning an incorrect value.  
```
```suggestion
- When using stacked charts, the macros `$sum` and `$unformattedSum` are now working properly for customized tooltips.Previously, the `$sum` and `$unformattedSum` macros were returning an incorrect value for zero-valued data plot.
```

Eric-revollo replied:
changed

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+- When using stacked charts, the macros `$sum` and `$unformattedSum` are now working properly for customized tooltips.Previously, the `$sum` and `$unformattedSum` macros were returning an incorrect value.  
+
+- The "%" symbol is now displayed properly in situations where the trendline value coincides with the div line value of the yAxis of 100percent stacked charts. Earlier the "%" symbol was not visible in such situations. 
```
```suggestion
- The `%` symbol is now displayed properly in situations where the trendline value coincides with the yAxis div line value for 100percent stacked charts. Earlier the `%` symbol was not visible in such situations.
```

Eric-revollo replied:
changed

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+
+- The attribute `seriesname` of a dataset is now working properly for charts rendered in IE 11 browser. Earlier, the usage of 1 {br} tag was considered as 2 {br} tags in IE 11 browser.
+
+- The event handlers `chartRollOver` and `chartRollOut`are now working properly when multiple charts are present on a web page. Earlier, upon multiple instances of hover over each other one of the charts stopped invoking the hover events and threw a JS error. 
```
This is slightly incorrect. The situation was when multiple charts are present on page web with `chartRollOver` and `chartRollOut` events handled by a common handler function then upon multiple instances of hover charts threw JS error and stopped invoking the events.

Eric-revollo replied:
changed

sikrigagan commented on upgrading/change-log.md:
```diff
@@ -7,6 +7,39 @@ heading: Version 3.15.x
+- The "%" symbol is now displayed properly in situations where the trendline value coincides with the div line value of the yAxis of 100percent stacked charts. Earlier the "%" symbol was not visible in such situations. 
+
+
+<h4>Improvements</h4>
```
FC 8 (RED-8906) is missing, please add it in improvements section.

sikrigagan replied:
@Eric-revollo This has been added to **Fixes** section, please move it to **Improvements** section.

Eric-revollo commented:
FC3: Fixed, need better data sample.
FC4: Fixed, not sure how can we add an interactive example with css added a fidle
FC5: Fixed, example should be set on 1
FC7: seems to be some confusion with FC8?
FC11: Fixed
FC12: Added to changelog,
_"@sikrigagan if I'm not mistaken link related events are exclusive to Chord. If yes, then do you agree that we should have a separate section on events page for Chord related events"_  do we have a decision on this?
Still will need help with samples.

FusionTime
FT2 and FT3: Fixed
FT4: Fixed
FT5: Fixed

niharsawant commented on common-use-cases/css-transformations.md:
```diff
@@ -39,4 +39,4 @@ You can apply the transformations on the parent object of a chart.
 Here is an exmaple (refer to the HTML tab to see the source code):
-<iframe width="100%" height="600" src="//jsfiddle.net/fusioncharts/hzc1rLxa/embedded/result,html,js/" allowfullscreen="allowfullscreen" allowpaymentrequest frameborder="0"></iframe>
\ No newline at end of file
+<iframe width="100%" height="600" src="//jsfiddle.net/fusioncharts/hzc1rLxa/4/embedded/result,html,js/" allowfullscreen="allowfullscreen" allowpaymentrequest frameborder="0"></iframe>
```
I had updated the base version of the main fiddle itself hence it was not needed to provide the version. If tomorrow, we change the fiddle again we don't have to make the changes here. That was the idea. But it seems OK. Not changing it back.

niharsawant commented on upgrading/whats-new.md:
```diff
@@ -10,7 +10,7 @@ This section is for users who are using the previous version of FusionCharts in
 Expanded support for click and hover events node, ribbon, and labels; as well as link attributes for node, ribbon, and labels. Refer to [Chord Diagram documentation](/chart-guide/standard-charts/chord-diagram) to know more.
 
-Here is an exmaple, (hover over any chord node or links to see the events in action):
+Here is an example, (hover over any chord node or links to see the events in action):
```
Sorry :)

niharsawant commented:
@Eric-revollo I have made some changes in the FusionCharts side, please go through them. Let me know if there are any changes required (add an inline comment).
